### PR TITLE
Refactor out branches attribute to new data source

### PR DESCRIPTION
**Pull request by k24dizzle**
- Refactor out branches attribute from `resource_github_repository` and `data_source_github_repository`
- Introduce a new data_source called `data_source_github_repository_branches` that returns branches instead.

`Rationale:`
I believe branches shouldn't belong to the attributes of the `github_repository` resource/data source. These [fields](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/repository#attributes-reference) should be relevant to the repository itself, not other auxiliary resources (like teams, collaborators, branches, etc.)

Similar to [github_collaborators](https://registry.terraform.io/providers/integrations/github/latest/docs/data-sources/collaborators), [github_repository_pull_requests](https://registry.terraform.io/providers/integrations/github/latest/docs/data-sources/repository_pull_requests), and other data sources, it would be better to create a new data source to contain this information instead.

If a developer wants to interact with the branches of a repo, they can use `data_source_github_repository_branches` instead. This helps prevent some of the problems described in the issues below.

`Testing done in this repo:`
https://github.com/kzhaotest/branches

`Fixes:`
https://github.com/integrations/terraform-provider-github/issues/1037
https://github.com/integrations/terraform-provider-github/issues/1010

`While still supporting the functionality of:`
https://github.com/integrations/terraform-provider-github/issues/748

### CODE REVIEW

File: github/data_source_github_repository_branches.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+	return branchList
+}
+
+func dataSourceGithubRepositoryBranchesRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Owner).v3client
+	orgName := meta.(*Owner).name
+	repoName := d.Get("repository").(string)
+
+	branches, _, err := client.Repositories.ListBranches(context.TODO(), orgName, repoName, nil)
```
Comment (k24dizzle) [inline github/data_source_github_repository_branches.go:60]: note, this data source will only display the first 30 branches of a repository, this is the default setting of the [Github API endpoint](https://docs.github.com/en/rest/reference/branches#list-branches) 
Screen Shot 2022-04-18 at 4 25 30 PM

This mimics the previous behavior introduced in https://github.com/integrations/terraform-provider-github/pull/892: https://github.com/kzhaotest/branches_test1 (see in this repo although there are 50+ branches, only 30+ are shown in the README (tf located [here](https://github.com/kzhaotest/branches/tree/main/before))).

We can introduce pagination in a future PR, but for now just replicate the previous behavior

**Comment by k24dizzle** [comment]
Hi @kfcampbell, I'm curious if there's any thoughts on this/best ways to roll this out

**Comment by k24dizzle** [comment]
@kfcampbell friendly bump

**Comment by kfcampbell** [comment]
@k24dizzle apologies for the slow response time. I'll shoot to get this into the next release!

File: github/data_source_github_repository_branches.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package github
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-github/v43/github"
```
Comment (mareksapota-lyft) [inline github/data_source_github_repository_branches.go:7]:
```suggestion
	"github.com/google/go-github/v45/github"
```
To match the currently used version of `go-github`.

**Comment by DoctorPolski** [comment]
@kfcampbell it would be great to get this out ASAP if you can squeeze it in.

Thanks.

**Comment by kfcampbell** [comment]
@k24dizzle Apologies for the delay. Integration tests on this aren't passing for me: 

```
--- FAIL: TestAccGithubBranchesDataSource (83.24s)
    --- FAIL: TestAccGithubBranchesDataSource/manages_branches_of_a_new_repository (80.94s)
        --- SKIP: TestAccGithubBranchesDataSource/manages_branches_of_a_new_repository/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubBranchesDataSource/manages_branches_of_a_new_repository/with_an_individual_account (0.00s)
        --- FAIL: TestAccGithubBranchesDataSource/manages_branches_of_a_new_repository/with_an_organization_account (79.90s)
FAIL
```

<details open>
<summary>The error seems to be a result of this snippet:</summary>
<br>
    testing.go:654: Step 0 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        UPDATE: data.github_repository_branches.test
          branches:   "" => "<computed>"
          id:         "" => "<computed>"
          repository: "" => "tf-acc-test-branches-2wdmv"
        UPDATE: github_repository.test
          allow_auto_merge:       "false" => "false"
          allow_merge_commit:     "true" => "true"
          allow_rebase_merge:     "true" => "true"
          allow_squash_merge:     "true" => "true"
          archived:               "false" => "false"
          auto_init:              "true" => "true"
          default_branch:         "main" => "main"
          delete_branch_on_merge: "false" => "false"
          description:            "" => ""
          etag:                   "W/\"b693979949e92b7a9a6fe1a72dfb591c806dbf206830fb62131e9b10b6419376\"" => "W/\"b693979949e92b7a9a6fe1a72dfb591c806dbf206830fb62131e9b10b6419376\""
          full_name:              "kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv" => "kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv"
          git_clone_url:          "git://github.com/kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv.git" => "git://github.com/kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv.git"
          has_downloads:          "false" => "false"
          has_issues:             "false" => "false"
          has_projects:           "false" => "false"
          has_wiki:               "false" => "false"
          homepage_url:           "" => ""
          html_url:               "https://github.com/kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv" => "https://github.com/kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv"
          http_clone_url:         "https://github.com/kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv.git" => "https://github.com/kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv.git"
          id:                     "tf-acc-test-branches-2wdmv" => "tf-acc-test-branches-2wdmv"
          is_template:            "false" => "false"
          name:                   "tf-acc-test-branches-2wdmv" => "tf-acc-test-branches-2wdmv"
          node_id:                "R_kgDOH46FUQ" => "R_kgDOH46FUQ"
          pages.#:                "0" => "0"
          private:                "false" => "false"
          repo_id:                "529433937" => "529433937"
          ssh_clone_url:          "git@github.com:kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv.git" => "git@github.com:kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv.git"
          svn_url:                "https://github.com/kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv" => "https://github.com/kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv"
          template.#:             "0" => "0"
          visibility:             "public" => "public"
          vulnerability_alerts:   "true" => ""
        
        
        
        STATE:
        
        data.github_repository_branches.test:
          ID = kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv
          provider = provider.github
          branches.# = 1
          branches.0.name = main
          branches.0.protected = false
          repository = tf-acc-test-branches-2wdmv
        
          Dependencies:
            github_repository.test
        github_repository.test:
          ID = tf-acc-test-branches-2wdmv
          provider = provider.github
          allow_auto_merge = false
          allow_merge_commit = true
          allow_rebase_merge = true
          allow_squash_merge = true
          archived = false
          auto_init = true
          default_branch = main
          delete_branch_on_merge = false
          description = 
          etag = W/"b693979949e92b7a9a6fe1a72dfb591c806dbf206830fb62131e9b10b6419376"
          full_name = kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv
          git_clone_url = git://github.com/kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv.git
          has_downloads = false
          has_issues = false
          has_projects = false
          has_wiki = false
          homepage_url = 
          html_url = https://github.com/kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv
          http_clone_url = https://github.com/kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv.git
          is_template = false
          name = tf-acc-test-branches-2wdmv
          node_id = R_kgDOH46FUQ
          private = false
          repo_id = 529433937
          ssh_clone_url = git@github.com:kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv.git
          svn_url = https://github.com/kfcampbell-terraform-provider/tf-acc-test-branches-2wdmv
          visibility = public
          vulnerability_alerts = true
</details>

Are the tests passing for you locally?

File: github/data_source_github_repository_branches_test.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccGithubBranchesDataSource(t *testing.T) {
```
Comment (mareksapota-lyft) [inline github/data_source_github_repository_branches_test.go:11]:
```suggestion
func TestAccGithubRepositoryBranchesDataSource(t *testing.T) {
```

Comment (k24dizzle) [reply github/data_source_github_repository_branches_test.go]: ty :)

**Comment by mareksapota-lyft** [comment]
If dependabot is enabled on the organization by default tests will fail.  Tests pass with this setting off:

Screen Shot 2022-09-01 at 4 25 49 PM

https://github.com/organizations/ORGNAME/settings/security_analysis

**Pull request merged**